.github/workflows/auto-update-gemfiles.yml

```diff
@@ -0,0 +1,169 @@
+          path: gemfiles/${{ matrix.engine.name }}_${{ matrix.engine.version }}_*
+          retention-days: 3
+
+  aggregate:
+    needs: build
+    runs-on: ubuntu-latest
+    permissions:
+      actions: read # read secrets
+      pull-requests: write
+      contents: write
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+
+      - name: Download artifacts for Ruby 2.5
+        uses: actions/download-artifact@v4
+        with:
+          name: 'datadog-gem-ruby-2.5-gha${{ github.run_id }}'
+          path: gemfiles
+
+      - name: Download artifacts for Ruby 2.6
+        uses: actions/download-artifact@v4
+        with:
+          name: 'datadog-gem-ruby-2.6-gha${{ github.run_id }}'
+          path: gemfiles
+
+      - name: Download artifacts for Ruby 2.7
+        uses: actions/download-artifact@v4
+        with:
+          name: 'datadog-gem-ruby-2.7-gha${{ github.run_id }}'
+          path: gemfiles
+
+      - name: Download artifacts for Ruby 3.0
+        uses: actions/download-artifact@v4
+        with:
+          name: 'datadog-gem-ruby-3.0-gha${{ github.run_id }}'
+          path: gemfiles
+
+      - name: Download artifacts for Ruby 3.1
+        uses: actions/download-artifact@v4
+        with:
+          name: 'datadog-gem-ruby-3.1-gha${{ github.run_id }}'
+          path: gemfiles
+
+      - name: Download artifacts for Ruby 3.2
+        uses: actions/download-artifact@v4
+        with:
+          name: 'datadog-gem-ruby-3.2-gha${{ github.run_id }}'
+          path: gemfiles
+
+      - name: Download artifacts for Ruby 3.3
+        uses: actions/download-artifact@v4
+        with:
+          name: 'datadog-gem-ruby-3.3-gha${{ github.run_id }}'
+          path: gemfiles
+
+      - name: Download artifacts for Ruby 3.4
+        uses: actions/download-artifact@v4
+        with:
+          name: 'datadog-gem-ruby-3.4-gha${{ github.run_id }}'
+          path: gemfiles
+
+      - name: Download artifacts for JRuby 9.2
+        uses: actions/download-artifact@v4
+        with:
+          name: 'datadog-gem-jruby-9.2-gha${{ github.run_id }}'
+          path: gemfiles
+      
+      - name: Download artifacts for JRuby 9.3
+        uses: actions/download-artifact@v4
+        with:
+          name: 'datadog-gem-jruby-9.3-gha${{ github.run_id }}'
+          path: gemfiles
+
+      - name: Download artifacts for JRuby 9.4
+        uses: actions/download-artifact@v4
+        with:
+          name: 'datadog-gem-jruby-9.4-gha${{ github.run_id }}'
+          path: gemfiles
+      
+      # ADD NEW RUBIES HERE
+      
+      - run: git diff
+
+      - name: Create Pull Request
+        uses: peter-evans/create-pull-request@v7
+        with:
+          token: ${{secrets.GITHUB_TOKEN}}
+          branch: update-latest-gemfiles
+          commit-message: "update latest gemfiles"
+          delete-branch: true
+          base: master
+          title: 'Update Latest Gemfiles'
+          body: |
```

Comment:
No need to follow the template structure, refining the body: 

Describing itself and origin, for example:
> This is an Autogenerated PR from blablabla workflow

Explaining the intention, for example:
> This PR updates our integration dependencies being tested so we can keep up with the community development

Call to action, for example
> Merge it if green, if not .....


In addition: apply the correct label and staff to review the PR